front/lib/api/assistant/actions/conversation/list_files.ts

```diff
@@ -0,0 +1,105 @@
+import type {
+  AgentMessageType,
+  ConversationFileType,
+  ConversationListFilesActionType,
+  ConversationType,
+  FunctionCallType,
+  FunctionMessageTypeModel,
+  ModelId,
+} from "@dust-tt/types";
+import {
+  BaseAction,
+  getTablesQueryResultsFileTitle,
+  isAgentMessageType,
+  isContentFragmentType,
+  isTablesQueryActionType,
+} from "@dust-tt/types";
+
+interface ConversationListFilesActionBlob {
+  agentMessageId: ModelId;
+  functionCallId: string | null;
+  functionCallName: string | null;
+  files: ConversationFileType[];
+}
+
+export class ConversationListFilesAction extends BaseAction {
+  readonly agentMessageId: ModelId;
+  readonly files: ConversationFileType[];
+  readonly functionCallId: string | null;
+  readonly functionCallName: string | null;
+  readonly step: number = -1;
+  readonly type = "conversation_list_files_action";
+
+  constructor(blob: ConversationListFilesActionBlob) {
+    super(-1, "conversation_list_files_action");
+
+    this.agentMessageId = blob.agentMessageId;
+    this.files = blob.files;
+    this.functionCallId = blob.functionCallId;
+    this.functionCallName = blob.functionCallName;
+  }
+
+  renderForFunctionCall(): FunctionCallType {
+    return {
+      id: this.functionCallId ?? `call_${this.id.toString()}`,
+      name: this.functionCallName ?? "list_conversation_files",
+      arguments: JSON.stringify({}),
+    };
+  }
+
+  renderForMultiActionsModel(): FunctionMessageTypeModel {
+    let content = "CONVERSATION FILES:\n";
+    for (const f of this.files) {
+      content += `<file id="${f.fileId}" name="${f.title}" type="${f.contentType}" />\n`;
+    }
+
+    return {
+      role: "function" as const,
+      name: this.functionCallName ?? "list_conversation_files",
+      function_call_id: this.functionCallId ?? `call_${this.id.toString()}`,
+      content,
+    };
+  }
+}
+
+export function makeConversationListFilesAction(
+  agentMessage: AgentMessageType,
+  conversation: ConversationType
+): ConversationListFilesActionType | null {
+  const jitFiles: ConversationFileType[] = [];
```

Comment:
nit: "jitFiles"  could really just be "files". They aren't really "jit" IMHO

Reply:
Good catch. Reminiscence from early versions. Will fix.